packages/themes/build.js

```diff
@@ -0,0 +1,243 @@
+const tokenMatchesAnyCondition = (token, conditions) => {
+  return conditions.some((condition) => matchesCondition(token, condition));
+};
+
+const createMatcher = (conditions) => (token) => tokenMatchesAnyCondition(token, conditions);
+
+// REGISTER THE CUSTOM FILTERS USING OUR MATCHING CONDITIONS
+
+StyleDictionary.registerFilter({
+  name: 'cssGlobalFilter',
+  matcher: (token) => createMatcher(conditionsCssGlobal)(token)
+});
+
+StyleDictionary.registerFilter({
+  name: 'cssLightThemeFilter',
+  matcher: (token) => createMatcher(conditionsCssLightTheme)(token)
+});
+
+StyleDictionary.registerFilter({
+  name: 'cssDarkThemeFilter',
+  matcher: (token) => createMatcher(conditionsCssDarkTheme)(token)
+});
+
+StyleDictionary.registerFilter({
+  name: 'twFilter',
+  matcher: (token) => createMatcher(conditionsTw)(token)
+});
+
+StyleDictionary.registerFilter({
+  name: 'jsFilter',
+  matcher: (token) => createMatcher(conditionsJs)(token)
+});
+
+/**
+ * Custom format that handle reference in css variables
+ */
+const transformString = (str, replace = 'theme', regex = /.*-color/) => {
+  // Use a regular expression to replace everything before regex with replace
+  return str.replace(regex, replace);
+};
+
+StyleDictionary.registerFormat({
+  name: 'css/variables',
+  formatter({ dictionary }) {
+    return `${this.options.selector} {
+          ${dictionary.allTokens
+            .map((token) => {
+              let originalName = token.name;
+
+              let themedName =
+                token.path[0] === 'global'
+                  ? originalName
+                  : transformString(originalName, 'theme-', /.*-color-/);
+
+              if (dictionary.usesReference(token.original.value)) {
+                const reference = dictionary.getReferences(token.original.value);
+                const referenceName =
+                  reference[0].path[0] === 'global'
+                    ? reference[0].name
+                    : transformString(reference[0].name);
+                return `  --${themedName}: var(--${referenceName}, ${token.value});`;
+              }
+
+              return `  --${themedName}: ${token.value};`;
+            })
+            .join('\n')}
+        }`;
+  }
+});
+
+/**
+ * Custom format that generates tailwind color config based on css variables
+ */
+
+StyleDictionary.registerFormat({
+  name: 'tw/css-variables',
+  formatter({ dictionary }) {
+    return (
+      'module.exports = ' +
+      `{\n${dictionary.allTokens
+        .map((token) => {
+          let originalName = token.name;
+
+          let themedName =
+            token.path[0] === 'global'
+              ? transformString(originalName, 'global-color')
+              : transformString(originalName, 'theme-', /.*-color-/);
+
+          let transformedName = transformString(originalName, 'color-', /.*-color-/);
+
+          return `  "${transformedName}": "var(--${themedName}, ${token.value})"`;
+        })
+        .join(',\n')}\n}`
+    );
+  }
+});
```

Comment:
Again, I'd extract the anonymous function out of the template string:


```suggestion

const formatTailwindColor = (token) => {
    let originalName = token.name;

    let themedName =
      token.path[0] === 'global'
        ? transformString(originalName, 'global-color')
        : transformString(originalName, 'theme-', /.*-color-/);

    let transformedName = transformString(originalName, 'color-', /.*-color-/);

    return `  "${transformedName}": "var(--${themedName}, ${token.value})"`;
}

StyleDictionary.registerFormat({
  name: 'tw/css-variables',
  formatter({ dictionary }) {
    return (
      'module.exports = ' +
      `{\n${dictionary.allTokens.map(formatTailwindColor).join(',\n')}\n}`
    );
  }
});
```

Also, you can use a template string instead of the string concatenation:

```js
'module.exports = ' +
`{\n${dictionary.allTokens.map(formatTailwindColor).join(',\n')}\n}`
```

```js
`module.exports = {
  ${dictionary.allTokens.map(formatTailwindColor).join(',\n')}
}
`
```